gammapy/maps/wcsnd.py

```diff
@@ -495,38 +498,116 @@ def plot(self, ax=None, fig=None, add_cbar=False, stretch='linear', **kwargs):
         import matplotlib.pyplot as plt
         from astropy.visualization import simple_norm
 
-        if not self.geom.is_image:
-            raise TypeError('Only supported on 2D maps')
 
         if fig is None:
             fig = plt.gcf()
 
-        if ax is None:
-            ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
-
-        data = self.data
 
         kwargs.setdefault('interpolation', 'nearest')
         kwargs.setdefault('origin', 'lower')
         kwargs.setdefault('cmap', 'afmhot')
-        norm = simple_norm(data[np.isfinite(data)], stretch)
-        kwargs.setdefault('norm', norm)
 
-        caxes = ax.imshow(data, **kwargs)
 
-        cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
-        try:
-            ax.coords['glon'].set_axislabel('Galactic Longitude')
-            ax.coords['glat'].set_axislabel('Galactic Latitude')
-        except KeyError:
-            ax.coords['ra'].set_axislabel('Right Ascension')
-            ax.coords['dec'].set_axislabel('Declination')
-        except AttributeError:
-            log.info("Can't set coordinate axes. No WCS information available.")
-
-        # without this the axis limits are changed when calling scatter
-        ax.autoscale(enable=False)
-        return fig, ax, cbar
+
+        if self.geom.ndim == 2:
+            if ax is None:
+                ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
+
+            data = self.data
+            norm = simple_norm(data[np.isfinite(data)], stretch)
+            kwargs.setdefault('norm', norm)
+            caxes = ax.imshow(data, **kwargs)
+            cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
+            # without this the axis limits are changed when calling scatter
+            ax.autoscale(enable=False)
+
+            try:
+                ax.coords['glon'].set_axislabel('Galactic Longitude')
+                ax.coords['glat'].set_axislabel('Galactic Latitude')
+            except KeyError:
+                ax.coords['ra'].set_axislabel('Right Ascension')
+                ax.coords['dec'].set_axislabel('Declination')
+            except AttributeError:
+                log.info("Can't set coordinate axes. No WCS information available.")
+
+            return fig, ax, cbar
+
+        elif self.geom.ndim == 3:
+
+            try:
+                from ipywidgets.widgets.interaction import interact, fixed
+                import ipywidgets as widgets
+            except:
+                raise ImportError('ipywidgets is not installed.')
+
+            @interact(
+                     index=widgets.IntSlider(min=0, max=self.data.shape[0] - 1, step=1, value=1,
```

Comment:
Can it be possible to have as many sliders as non-spatial axis?
And what will be plotted is passed to `_plot_interactive` is `slice` instead of `index`
where `slice` is the list of sliders. Then this will work for 4-d cubes as well, and the user can easily see if all the slices are proper...

Reply:
I also like this suggestion (as alternative to give the axis name in the plot method), but then the spectrum / profile plot has to be removed for sure (which is fine by me...).